### PR TITLE
maint: update brod dep to 4.0

### LINKED ISSUE #144: Fails to compile on otp 27
Hi, switching to elixir 1.17.0 and otp 27 breaks this for me.

The setup:

asdf .tool_versions
```
elixir 1.17.0-otp-27
erlang 27.0
```

asdf install
mix deps.get
mix deps.compile

results in:
```
** (Mix) Could not compile dependency :snappyer, "/home/gregor/.asdf/installs/elixir/1.17.0-otp-27/.mix/elixir/1-17/rebar3 bare compile --paths .../_build/dev/lib/*/ebin" command failed. Errors may have been logged above. You can recompile this dependency with "mix deps.compile snappyer --force", update it with "mix deps.update snappyer" or clean it with "mix deps.clean snappyer"
```

Obviously, this is not an issue with broadway_kafka directly, but with snappyer, which is a dependency of brod. The reason I'm opening an issue here is that both brod and snappyer are erlang libs and I'm not sure how to reproduce this without mix.

### PR DESCRIPTION
**Pull request by deepfryed**
Updates brod dep to 4.0, should fix #144

### CODE REVIEW

**Comment by josevalim** [comment]
Can you please run `mix deps.unlock —unused` as well?

File: mix.exs [hunk 1]
```diff
@@ -27,7 +27,7 @@ defmodule BroadwayKafka.MixProject do
   defp deps do
     [
       {:broadway, "~> 1.0"},
-      {:brod, "~> 3.16"},
+      {:brod, "~> 4.0"},
```
Comment (josevalim) [inline mix.exs:30]:
```suggestion
      {:brod, "~> 3.16 or ~> 4.0"},
```

Comment (deepfryed) [reply mix.exs]: done, thanks

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by deepfryed** [comment]
thanks, appreciate the quick review / feedback cycle.

**Comment by slashmili** [comment]
since we support both brod versions, I'd say let's push it to hex.

For the record the integration test failed on main but passed after I re-ran it today!

**Comment by esambo** [comment]
BTW, it looks like 93fdc70f258c9a95f8ff9097df7d4d94c373959b that has tag `v0.4.4` hasn't been merged to master. GitHub shows an error with:
> This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.

**Comment by josevalim** [comment]
Pushed.

**Pull request merged**